python/pyproject.toml

```diff
@@ -27,7 +27,7 @@ requires = [
     # configuring setuptools_scm in pyproject.toml requires
     # versions released after 2022
     "setuptools_scm[toml]>=8",
-    "setuptools>=64",
+    "setuptools>=77",
```

Comment:
This is required due to setuptools<77 not recognizing the new license strings proposed in this PR (see https://github.com/pypa/setuptools/issues/4903 for explanation). I'm not sure if this is controversial so bringing it up here. @raulcd